### ISSUE TITLE
[JENKINS-57371] error message when building PR

### ISSUE THREAD

**Issue by jenkins-infra-bot** [jenkinsci/github-branch-source-plugin#1257]
<p>Hi,</p>

<p> </p>

<p>After updating to the latest version (2.5.1) I get the following error messages immediately and the build fails.</p>

<p> </p>

<p>ERROR: Invalid merge hash for pull request 522 : Head and base commits do match merge commit a538d719213e99bb52dc73c2fac2fd3b1375ee1d+7b4572c5176951887e3fd3cf80df30821caed9ec (8f660b9de0bb201869379a428d117866539f4cce)<br/>
Finished: FAILURE</p>

---
<details><summary><i>Originally reported by <img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/nickez.jpg" title="nickez's avatar" /> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=nickez">nickez</a>, imported from: <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-57371" target="_blank">error message when building PR</a></i></summary>
<i><ul>
<li><b>assignee</b>: <img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/bitwiseman.png" title="bitwiseman's avatar" /> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=bitwiseman">bitwiseman</a>
<li><b>status</b>: Closed
<li><b>priority</b>: Major
<li><b>component(s)</b>: github-branch-source-plugin
<li><b>resolution</b>: Fixed
<li><b>resolved</b>: 2019-05-24T16:09:55+00:00
<li><b>votes</b>: 1
<li><b>watchers</b>: 4
<li><b>imported</b>: 2025-12-02
</ul></i>
<details><summary>Raw content of original issue</summary>

<pre>
<p>Hi,</p>

<p> </p>

<p>After updating to the latest version (2.5.1) I get the following error messages immediately and the build fails.</p>

<p> </p>

<p>ERROR: Invalid merge hash for pull request 522 : Head and base commits do match merge commit a538d719213e99bb52dc73c2fac2fd3b1375ee1d+7b4572c5176951887e3fd3cf80df30821caed9ec (8f660b9de0bb201869379a428d117866539f4cce)
Finished: FAILURE</p></pre>
</details>
</details>

**Comment by jenkins-infra-bot** [comment]
- _Remote link associated with this issue:_


  - [PR-226](https://github.com/jenkinsci/github-branch-source-plugin/pull/226)

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/bitwiseman.png" title="bitwiseman's avatar" /> bitwiseman:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-57371?focusedId=366427&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-366427">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=nickez" class="user-hover" rel="nickez">nickez</a>
The message indicate that the merge commit for this PR has parents other than commits that are at the head of the source and target branches. </p>

<p>Please upgrade to 2.5.2 (just released).   While it will probably get the same result, but the output maybe different.</p>

<p>Is this an older PR? 
(If public) What repo is this on? 
what is the result from <a href="https://api.github.com/repos/YOURORG/YOURREPO/commits/8f660b9de0bb201869379a428d117866539f4cce" class="external-link" target="_blank" rel="nofollow noopener">https://api.github.com/repos/YOURORG/YOURREPO/commits/8f660b9de0bb201869379a428d117866539f4cce</a></p>

<p>Are you getting this message for multiple PRs?  
Does this happen after you run a scan? </p>
</pre>
</details>

</ul>
</details>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=nickez" class="user-hover" rel="nickez">nickez</a><br/>
The message indicate that the merge commit for this PR has parents other than commits that are at the head of the source and target branches. </p>

<p>Please upgrade to 2.5.2 (just released).   While it will probably get the same result, but the output maybe different.</p>

<p>Is this an older PR? <br/>
(If public) What repo is this on? <br/>
what is the result from <a href="https://api.github.com/repos/YOURORG/YOURREPO/commits/8f660b9de0bb201869379a428d117866539f4cce" class="external-link" target="_blank" rel="nofollow noopener">https://api.github.com/repos/YOURORG/YOURREPO/commits/8f660b9de0bb201869379a428d117866539f4cce</a></p>

<p>Are you getting this message for multiple PRs?  <br/>
Does this happen after you run a scan? </p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/nickez.jpg" title="nickez's avatar" /> nickez:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-57371?focusedId=366453&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-366453">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Right now I'm a little bit hesitant to update again because last time this issue occurred and broke our CI.</p>

<p>It was a PR from a fork of a private repo. And I tried opening multiple PRs with all the same issue.</p>

<p>I didn't try to rescan the repository. Is that necessary every time I update the plugin?</p>
</pre>
</details>

</ul>
</details>

<p>Right now I'm a little bit hesitant to update again because last time this issue occurred and broke our CI.</p>

<p>It was a PR from a fork of a private repo. And I tried opening multiple PRs with all the same issue.</p>

<p>I didn't try to rescan the repository. Is that necessary every time I update the plugin?</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/bitwiseman.png" title="bitwiseman's avatar" /> bitwiseman:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-57371?focusedId=366494&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-366494">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=nickez" class="user-hover" rel="nickez">nickez</a>
I certainly don't want you to break your CI, but there isn't enough information here to be able to debug this. </p>

<p>The rescan is not generally needed, I'm just trying to get more information and see if there's a reasonable workaround. </p>

<p>To explain more fully: this failure occurs when the plugin detects that the merge_commit_sha provided by github is not the immediate child of the head commit of the source branch and the head commit of the base branch. This can happen when the base branch is updated after the PR job is queued but before the PR starts.  Usually that would mean another PR job is also queued after this one and it will have consistent values and run successfully.  It is a safety measure to ensure accurate CI runs.</p>

<p>That fact that you're getting this (even on new PRs) indicates something is changing the commit structure (or at least that Jenkins believes they are changing).  </p>

<p>This definitely is not happening to everyone - 2.5.1 was confirmed to work by several reporters of a previous regression.  So, there's something different about your configuration. We need to figure out what that is before we can address/fix it. </p>

<p>Questions:
Is this happen on jobs triggered by webhook?
What happens when you do a scan of the repo?  (Can you provide the scan log?)
For an open PR, what happens when you manually run the pipeline job? 
(Double checking) Did this error happen on new PR's that you opened after the upgrade? </p>

<p>Are you doing anything of note to your repo (pushing or amending commits) as you start PRs?</p>

<p>(Double checking)You tried closing and reopening the PR on GitHub? When you did that were the commit sha values in the error different after close and reopen?</p>

<p>For each of the sha values in that error line I need to see the result of the following:
what is the result from this api query:
<a href="https://api.github.com/repos/YOURORG/YOURREPO/commits/COMMITSHA" class="external-link" target="_blank" rel="nofollow noopener">https://api.github.com/repos/YOURORG/YOURREPO/commits/COMMITSHA</a></p>

<p>I could also use the info from: 
<a href="https://api.github.com/repos/YOURORG/YOURREPO/pulls/PULLNUMBER" class="external-link" target="_blank" rel="nofollow noopener">https://api.github.com/repos/YOURORG/YOURREPO/pulls/PULLNUMBER</a></p>

<p>For a private repo, you'll need pass credentials or an api key to access these. </p>

<p>If you'd rather chat on <a href="https://gitter.im/bitwiseman" class="external-link" target="_blank" rel="nofollow noopener">https://gitter.im/bitwiseman</a> (or some other chat platform), we can do that too. </p>
</pre>
</details>

</ul>
</details>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=nickez" class="user-hover" rel="nickez">nickez</a><br/>
I certainly don't want you to break your CI, but there isn't enough information here to be able to debug this. </p>

<p>The rescan is not generally needed, I'm just trying to get more information and see if there's a reasonable workaround. </p>

<p>To explain more fully: this failure occurs when the plugin detects that the merge_commit_sha provided by github is not the immediate child of the head commit of the source branch and the head commit of the base branch. This can happen when the base branch is updated after the PR job is queued but before the PR starts.  Usually that would mean another PR job is also queued after this one and it will have consistent values and run successfully.  It is a safety measure to ensure accurate CI runs.</p>

<p>That fact that you're getting this (even on new PRs) indicates something is changing the commit structure (or at least that Jenkins believes they are changing).  </p>

<p>This definitely is not happening to everyone - 2.5.1 was confirmed to work by several reporters of a previous regression.  So, there's something different about your configuration. We need to figure out what that is before we can address/fix it. </p>

<p>Questions:<br/>
Is this happen on jobs triggered by webhook?<br/>
What happens when you do a scan of the repo?  (Can you provide the scan log?)<br/>
For an open PR, what happens when you manually run the pipeline job? <br/>
(Double checking) Did this error happen on new PR's that you opened after the upgrade? </p>

<p>Are you doing anything of note to your repo (pushing or amending commits) as you start PRs?</p>

<p>(Double checking)You tried closing and reopening the PR on GitHub? When you did that were the commit sha values in the error different after close and reopen?</p>

<p>For each of the sha values in that error line I need to see the result of the following:<br/>
what is the result from this api query:<br/>
<a href="https://api.github.com/repos/YOURORG/YOURREPO/commits/COMMITSHA" class="external-link" target="_blank" rel="nofollow noopener">https://api.github.com/repos/YOURORG/YOURREPO/commits/COMMITSHA</a></p>

<p>I could also use the info from: <br/>
<a href="https://api.github.com/repos/YOURORG/YOURREPO/pulls/PULLNUMBER" class="external-link" target="_blank" rel="nofollow noopener">https://api.github.com/repos/YOURORG/YOURREPO/pulls/PULLNUMBER</a></p>

<p>For a private repo, you'll need pass credentials or an api key to access these. </p>

<p>If you'd rather chat on <a href="https://gitter.im/bitwiseman" class="external-link" target="_blank" rel="nofollow noopener">https://gitter.im/bitwiseman</a> (or some other chat platform), we can do that too. </p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>brandonlesann:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-57371?focusedId=366530&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-366530">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>I had the exact same issue as <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=nickez" class="user-hover" rel="nickez">nickez</a> . </p>

<p>Closing and reopening the PR did not work. Creating new different PRs still produced the same error (although with different hash values). </p>

<p>Ignoring this error message and merging the PR solved the issue and doesn't seem to appear again...</p>

<p>Hope it helps.</p>
</pre>
</details>

</ul>
</details>

<p>I had the exact same issue as <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=nickez" class="user-hover" rel="nickez">nickez</a> . </p>

<p>Closing and reopening the PR did not work. Creating new different PRs still produced the same error (although with different hash values). </p>

<p>Ignoring this error message and merging the PR solved the issue and doesn't seem to appear again...</p>

<p>Hope it helps.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/bitwiseman.png" title="bitwiseman's avatar" /> bitwiseman:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-57371?focusedId=366611&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-366611">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=brandonlesann" class="user-hover" rel="brandonlesann">brandonlesann</a></p>
<blockquote><p>Ignoring this error message and merging the PR solved the issue and doesn't seem to appear again...</p></blockquote>

<p>What do you mean here? Once you merge one PR the rest of the PRs work okay from then on?  Could you be a bit more specific? </p>
</pre>
</details>

</ul>
</details>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=brandonlesann" class="user-hover" rel="brandonlesann">brandonlesann</a></p>
<blockquote><p>Ignoring this error message and merging the PR solved the issue and doesn't seem to appear again...</p></blockquote>

<p>What do you mean here? Once you merge one PR the rest of the PRs work okay from then on?  Could you be a bit more specific? </p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/nickez.jpg" title="nickez's avatar" /> nickez:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-57371?focusedId=366697&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-366697">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>I updated to 2.5.2 and cannot reproduce this issue any more. Thanks!</p>
</pre>
</details>

</ul>
</details>

<p>I updated to 2.5.2 and cannot reproduce this issue any more. Thanks!</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>brandonlesann:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-57371?focusedId=366728&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-366728">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=bitwiseman" class="user-hover" rel="bitwiseman">bitwiseman</a> , yes, once the PR was merged the next PRs worked fine. </p>

<p> </p>

<p><cite>Once you merge one PR the rest of the PRs work okay from then on?  </cite></p>

<p>exactly. When the first PR was merged, the other PRs were rebuilt and were no longer throwing this error.</p>

<p>And while this error was still present, the build of the other branches worked fine. It was just the PRs failing. </p>
</pre>
</details>

</ul>
</details>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=bitwiseman" class="user-hover" rel="bitwiseman">bitwiseman</a> , yes, once the PR was merged the next PRs worked fine. </p>

<p> </p>

<p><cite>Once you merge one PR the rest of the PRs work okay from then on?  </cite></p>

<p>exactly. When the first PR was merged, the other PRs were rebuilt and were no longer throwing this error.</p>

<p>And while this error was still present, the build of the other branches worked fine. It was just the PRs failing. </p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/bitwiseman.png" title="bitwiseman's avatar" /> bitwiseman:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-57371?focusedId=366780&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-366780">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=nickez" class="user-hover" rel="nickez">nickez</a>
Excellent!  </p>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=brandonlesann" class="user-hover" rel="brandonlesann">brandonlesann</a>
Yes, the only changes in 2.5.x were ones that changed the way PRs were handled.  So, branches not being effected makes sense. 
The behavior you're seeing is truly odd.  But if it is all clear now, that is the important part.  </p>

<p>I'm going to leave this open in case there are other who see upgrade issues to v2.5.2. </p>
</pre>
</details>

</ul>
</details>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=nickez" class="user-hover" rel="nickez">nickez</a><br/>
Excellent!  </p>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=brandonlesann" class="user-hover" rel="brandonlesann">brandonlesann</a><br/>
Yes, the only changes in 2.5.x were ones that changed the way PRs were handled.  So, branches not being effected makes sense. <br/>
The behavior you're seeing is truly odd.  But if it is all clear now, that is the important part.  </p>

<p>I'm going to leave this open in case there are other who see upgrade issues to v2.5.2. </p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>audunhalland:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-57371?focusedId=367066&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-367066">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>This also happens to us. I'm getting error message:
ERROR: Invalid merge hash for pull request 47 : Head and base commits do match merge commit 5729a1e31f5d0691d16801604312de0af03da009+7e881478aa951a0f9caad966907c68c5c058a935 (98e8973772d6f2075ea775f7e985e0a739e3fc34)
Finished: FAILURE.</p>

<p>5729 is the tip of the PR branch. 7e88 is the commit of the <em>previously merged</em> PR (onto master). However, our pipeline generates and pushes a release commit to github for the master job, where it updates the version number of the app, and tags that commit with the release number. This commit is set up to be ignored by jenkins (using Ignore Committer Strategy). So, the current tip of  master branch is not 7e88 anymore, and I suspect that this has something to do with this problem.</p>
</pre>
</details>

</ul>
</details>

<p>This also happens to us. I'm getting error message:<br/>
ERROR: Invalid merge hash for pull request 47 : Head and base commits do match merge commit 5729a1e31f5d0691d16801604312de0af03da009+7e881478aa951a0f9caad966907c68c5c058a935 (98e8973772d6f2075ea775f7e985e0a739e3fc34)<br/>
Finished: FAILURE.</p>

<p>5729 is the tip of the PR branch. 7e88 is the commit of the <em>previously merged</em> PR (onto master). However, our pipeline generates and pushes a release commit to github for the master job, where it updates the version number of the app, and tags that commit with the release number. This commit is set up to be ignored by jenkins (using Ignore Committer Strategy). So, the current tip of  master branch is not 7e88 anymore, and I suspect that this has something to do with this problem.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>audunhalland:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-57371?focusedId=367069&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-367069">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>However, I also found that "Scan Repository Now" does not see the pull request as updated, even if the tip of the PR branch is a merge that points to latest master. Log outputs:</p>

<p>Getting remote branches...</p>

<p>Checking branch master</p>

<p>Getting remote pull requests...
 ‘Jenkinsfile’ found
 Met criteria
 No changes detected: master (still at 76ff9c62d27d6b60666c08d0686e3100974fd1fe)</p>

<p>....</p>

<p>Checking pull request #47
 ‘Jenkinsfile’ found
 Met criteria
 No changes detected: PR-47 (still at 5729a1e31f5d0691d16801604312de0af03da009+7e881478aa951a0f9caad966907c68c5c058a935 (98e8973772d6f2075ea775f7e985e0a739e3fc34))</p>

<p>Current master is 76ff, so the PR checker should not have tried to use 5729+7e88, but 5729+76ff. Question remains whether this is misinformation from github, or incorrect logic in github-branch-source.</p>

<p>EDIT: PR strategy is "Merging the pull request with the current target branch revision".</p>
</pre>
</details>

</ul>
</details>

<p>However, I also found that "Scan Repository Now" does not see the pull request as updated, even if the tip of the PR branch is a merge that points to latest master. Log outputs:</p>

<p>Getting remote branches...</p>

<p>Checking branch master</p>

<p>Getting remote pull requests...<br/>
 ‘Jenkinsfile’ found<br/>
 Met criteria<br/>
 No changes detected: master (still at 76ff9c62d27d6b60666c08d0686e3100974fd1fe)</p>

<p>....</p>

<p>Checking pull request #47<br/>
 ‘Jenkinsfile’ found<br/>
 Met criteria<br/>
 No changes detected: PR-47 (still at 5729a1e31f5d0691d16801604312de0af03da009+7e881478aa951a0f9caad966907c68c5c058a935 (98e8973772d6f2075ea775f7e985e0a739e3fc34))</p>

<p>Current master is 76ff, so the PR checker should not have tried to use 5729+7e88, but 5729+76ff. Question remains whether this is misinformation from github, or incorrect logic in github-branch-source.</p>

<p>EDIT: PR strategy is "Merging the pull request with the current target branch revision".</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/bitwiseman.png" title="bitwiseman's avatar" /> bitwiseman:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-57371?focusedId=367152&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-367152">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=audunhalland" class="user-hover" rel="audunhalland">audunhalland</a></p>
<blockquote><p> This commit is set up to be ignored by jenkins (using Ignore Committer Strategy). So, the current tip of  master branch is not 7e88 anymore, and I suspect that this has something to do with this problem.</p></blockquote>

<p>This is extremely useful!  This could be where the problem is coming from. It is definitely some kind of incorrect assumptions made by github-branch-source. </p>

<p>Here's the problem: 
The plugin needs to make sure that the base, source, and merge commits are in sync otherwise the master (which uses the merge_commit) could be looking at different information from other agents, which use the source+base to create a new merge on the local file system.  From what you're describing, the ignore committer strategy forces plugin not to see the right base commit and so forces the commits out-of-sync. <img class="emoticon" src="https://issues.jenkins.io/images/icons/emoticons/sad.png" height="16" width="16" align="absmiddle" alt="" border="0"/>  </p>

<p>The merge_commit behavior was introduced in 2.5.0 - I was unaware of the filter trait you've mentioned.  I'm looking at it now. </p>
</pre>
</details>

</ul>
</details>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=audunhalland" class="user-hover" rel="audunhalland">audunhalland</a></p>
<blockquote><p> This commit is set up to be ignored by jenkins (using Ignore Committer Strategy). So, the current tip of  master branch is not 7e88 anymore, and I suspect that this has something to do with this problem.</p></blockquote>

<p>This is extremely useful!  This could be where the problem is coming from. It is definitely some kind of incorrect assumptions made by github-branch-source. </p>

<p>Here's the problem: <br/>
The plugin needs to make sure that the base, source, and merge commits are in sync otherwise the master (which uses the merge_commit) could be looking at different information from other agents, which use the source+base to create a new merge on the local file system.  From what you're describing, the ignore committer strategy forces plugin not to see the right base commit and so forces the commits out-of-sync. <img class="emoticon" src="https://issues.jenkins.io/images/icons/emoticons/sad.png" height="16" width="16" align="absmiddle" alt="" border="0"/>  </p>

<p>The merge_commit behavior was introduced in 2.5.0 - I was unaware of the filter trait you've mentioned.  I'm looking at it now. </p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/bitwiseman.png" title="bitwiseman's avatar" /> bitwiseman:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-57371?focusedId=367163&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-367163">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=audunhalland" class="user-hover" rel="audunhalland">audunhalland</a>
Wait a minute.  </p>

<p>If you run the PR-47 job manually, using "Build Now" instead of via the scan, does the Pipeline run successfully?
Also, what happens if you close and reopen that PR while making sure no other change goes into master? </p>

<p>I think there are two different issues converging here:
1. If the update of master to the new PR is causing the pipeline to fail due to inconsistent commits
2. The Ignore committer strategy is then preventing the PR from being recognized as having been updated.  </p>

<p>There are likely other strategies that will also cause this to fail. </p>

<p>Perhaps the right solution is to allow inconsistencies as long as the merge_commit is the child of two commits that are children of the two commits in the PR.  I'm not especially happy with that though. It feels like it opens the door to possible problems.   </p>

<p>I need to think about what the correct behavior should be in this case.  </p>
</pre>
</details>

</ul>
</details>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=audunhalland" class="user-hover" rel="audunhalland">audunhalland</a><br/>
Wait a minute.  </p>

<p>If you run the PR-47 job manually, using "Build Now" instead of via the scan, does the Pipeline run successfully?<br/>
Also, what happens if you close and reopen that PR while making sure no other change goes into master? </p>

<p>I think there are two different issues converging here:<br/>
1. If the update of master to the new PR is causing the pipeline to fail due to inconsistent commits<br/>
2. The Ignore committer strategy is then preventing the PR from being recognized as having been updated.  </p>

<p>There are likely other strategies that will also cause this to fail. </p>

<p>Perhaps the right solution is to allow inconsistencies as long as the merge_commit is the child of two commits that are children of the two commits in the PR.  I'm not especially happy with that though. It feels like it opens the door to possible problems.   </p>

<p>I need to think about what the correct behavior should be in this case.  </p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>audunhalland:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-57371?focusedId=367177&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-367177">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>"Build now" is as broken as the webhook( ? ) from github. Closing and re-opening the PR in github had no effect (although the PR was retained as a "disabled" Jenkins job in that meantime. I could try to delete it).</p>

<p>I can confirm that 7e88 is the first parent of master HEAD that is not ignored by Ignore Committer Strategy. And it is the first one that is a merge commit (a merge of the previous PR). The commits (two, in fact) at the tip of master, generated by jenkins, are regular commits.</p>
</pre>
</details>

</ul>
</details>

<p>"Build now" is as broken as the webhook( ? ) from github. Closing and re-opening the PR in github had no effect (although the PR was retained as a "disabled" Jenkins job in that meantime. I could try to delete it).</p>

<p>I can confirm that 7e88 is the first parent of master HEAD that is not ignored by Ignore Committer Strategy. And it is the first one that is a merge commit (a merge of the previous PR). The commits (two, in fact) at the tip of master, generated by jenkins, are regular commits.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/bitwiseman.png" title="bitwiseman's avatar" /> bitwiseman:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-57371?focusedId=367233&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-367233">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=audunhalland" class="user-hover" rel="audunhalland">audunhalland</a>
The fact that build now is broken is extremely odd.  Is it broken in the same way (same commits and message)? The place where "Build Now" calculates the HEAD of the base branch should be ignoring any filter.</p>

<p>Update: 
I've tested this with the Ignore Committer strategy and it works as expected: When I push a commit to master from a committer that should be ignored, the master branch doesn't build, but PR correctly detects the change and gets the most current commit from master which is the valid parent of the merge commit.  </p>

<p>What other plugins do you have configured?  
Also, out of curiosity are you using GitHub or GitHub Enterprise? </p>


</pre>
</details>

</ul>
</details>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=audunhalland" class="user-hover" rel="audunhalland">audunhalland</a><br/>
The fact that build now is broken is extremely odd.  Is it broken in the same way (same commits and message)? The place where "Build Now" calculates the HEAD of the base branch should be ignoring any filter.</p>

<p>Update: <br/>
I've tested this with the Ignore Committer strategy and it works as expected: When I push a commit to master from a committer that should be ignored, the master branch doesn't build, but PR correctly detects the change and gets the most current commit from master which is the valid parent of the merge commit.  </p>

<p>What other plugins do you have configured?  <br/>
Also, out of curiosity are you using GitHub or GitHub Enterprise? </p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>audunhalland:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-57371?focusedId=367241&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-367241">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Regular github.com, our jenkins is set up to scan the github organization. The repo in question is a private repository.</p>

<p>According to the Organization configuration page, these settings are active:</p>

<p>Behaviours:</p>
<ul>
	<li>Discover branches, Strategy: Exclude branches that are also filed as PRs</li>
	<li>Discover pull requests from origin, Strategy: Merging the pull request with the current target branch revision</li>
	<li>Checkout over SSH, Credentials: ****</li>
</ul>


<p>Project recognizers:</p>
<ul>
	<li>Pipeline Jenkinsfile, Script Path: Jenkinsfile</li>
</ul>


<p>Build strategies:</p>
<ul>
	<li>Ignore committer strategy
	<ul>
		<li>Don't trigger builds for pushes by certain Git commit author (comma-delimited list of author emails): <span class="error">&#91;ci email address&#93;</span></li>
		<li>Allow builds when a changeset contains non-ignored author(s): ENABLED</li>
	</ul>
	</li>
</ul>





<p>I might also add that when PR-47 was initially created, its branch was not up to date with master, and master got merged into it later, after its jenkins job was created. The latest commit of that PR currently is merge commit 5729, referencing child commits d75b (change set) + 76ff (master).</p>
</pre>
</details>

</ul>
</details>

<p>Regular github.com, our jenkins is set up to scan the github organization. The repo in question is a private repository.</p>

<p>According to the Organization configuration page, these settings are active:</p>

<p>Behaviours:</p>
<ul>
	<li>Discover branches, Strategy: Exclude branches that are also filed as PRs</li>
	<li>Discover pull requests from origin, Strategy: Merging the pull request with the current target branch revision</li>
	<li>Checkout over SSH, Credentials: ****</li>
</ul>


<p>Project recognizers:</p>
<ul>
	<li>Pipeline Jenkinsfile, Script Path: Jenkinsfile</li>
</ul>


<p>Build strategies:</p>
<ul>
	<li>Ignore committer strategy
	<ul>
		<li>Don't trigger builds for pushes by certain Git commit author (comma-delimited list of author emails): <span class="error">&#91;ci email address&#93;</span></li>
		<li>Allow builds when a changeset contains non-ignored author(s): ENABLED</li>
	</ul>
	</li>
</ul>





<p>I might also add that when PR-47 was initially created, its branch was not up to date with master, and master got merged into it later, after its jenkins job was created. The latest commit of that PR currently is merge commit 5729, referencing child commits d75b (change set) + 76ff (master).</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/bitwiseman.png" title="bitwiseman's avatar" /> bitwiseman:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-57371?focusedId=367259&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-367259">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=audunhalland" class="user-hover" rel="audunhalland">audunhalland</a>
All that looks completely reasonable.</p>

<p>I've tried a creating a version that I think will unblock you. 
Try installing this hpi: <a href="https://repo.jenkins-ci.org/incrementals/org/jenkins-ci/plugins/github-branch-source/2.5.3-rc840.68a7449454be/" class="external-link" target="_blank" rel="nofollow noopener">https://repo.jenkins-ci.org/incrementals/org/jenkins-ci/plugins/github-branch-source/2.5.3-rc840.68a7449454be/</a></p>

<p>This version is much more accommodating - it will fall back to cloning locally on master if the merge_commit provided by github doesn't match up.  Please give this a try and see how it works for you.  </p>

<p>Please keep an eye on the scans and build logs and tell me what you see. </p>
</pre>
</details>

</ul>
</details>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=audunhalland" class="user-hover" rel="audunhalland">audunhalland</a><br/>
All that looks completely reasonable.</p>

<p>I've tried a creating a version that I think will unblock you. <br/>
Try installing this hpi: <a href="https://repo.jenkins-ci.org/incrementals/org/jenkins-ci/plugins/github-branch-source/2.5.3-rc840.68a7449454be/" class="external-link" target="_blank" rel="nofollow noopener">https://repo.jenkins-ci.org/incrementals/org/jenkins-ci/plugins/github-branch-source/2.5.3-rc840.68a7449454be/</a></p>

<p>This version is much more accommodating - it will fall back to cloning locally on master if the merge_commit provided by github doesn't match up.  Please give this a try and see how it works for you.  </p>

<p>Please keep an eye on the scans and build logs and tell me what you see. </p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>audunhalland:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-57371?focusedId=367277&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-367277">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Thank you!</p>

<p>Scan log:</p>

<p><tt>...</tt>
<tt>Checking pull request #47</tt></p>

<p><tt>‘Jenkinsfile’ found Invalid github merge_commit_sha for pull request 47 : Head and base commits do match merge commit 5729a1e31f5d0691d16801604312de0af03da009+7e881478aa951a0f9caad966907c68c5c058a935 (567c876ca2e53637b53edd7c4eedca77af15aa65), expected head and base 76ff9c62d27d6b60666c08d0686e3100974fd1fe+5729a1e31f5d0691d16801604312de0af03da009 Met criteria Changes detected: PR-47 (5729a1e31f5d0691d16801604312de0af03da009+7e881478aa951a0f9caad966907c68c5c058a935 (567c876ca2e53637b53edd7c4eedca77af15aa65) → 5729a1e31f5d0691d16801604312de0af03da009+7e881478aa951a0f9caad966907c68c5c058a935 (UNKNOWN_MERGE_STATE)) Scheduled build for branch: PR-47</tt></p>

<p>Build log (it now starts building the pipeline):</p>

<p><tt>Branch indexing</tt>
 <tt>21:04:15 Connecting to <a href="https://api.github.com/" class="external-link" target="_blank" rel="nofollow noopener">https://api.github.com</a> using ci@mydomain.com/******</tt>
 <tt>Invalid github merge_commit_sha for pull request 47 : Head and base commits do match merge commit 5729a1e31f5d0691d16801604312de0af03da009+7e881478aa951a0f9caad966907c68c5c058a935 (567c876ca2e53637b53edd7c4eedca77af15aa65), expected head and base 76ff9c62d27d6b60666c08d0686e3100974fd1fe+5729a1e31f5d0691d16801604312de0af03da009</tt>
 <tt>Checking out git git@github.com:MyOrganization/my-repo.git into /var/jenkins_home/workspace/my-repo_PR-47@script to read Jenkinsfile</tt>
 <tt>using credential github-sshkey</tt>
 <tt>Cloning the remote Git repository</tt>
 <tt>Cloning with configured refspecs honoured and without tags</tt>
 <tt>Cloning repository git@github.com:MyOrganization/my-repo.git</tt>
 {{ &gt; git init /var/jenkins_home/workspace/my-repo_PR-47@script # timeout=10}}
 <tt>Fetching upstream changes from git@github.com:MyOrganization/my-repo.git</tt>
 {{ &gt; git --version # timeout=10}}
 <tt>using GIT_SSH to set credentials MyJenkins@github.com SSH key</tt>
 {{ &gt; git fetch --no-tags --force --progress git@github.com:MyOrganization/my-repo.git +refs/pull/47/head:refs/remotes/origin/PR-47 +refs/heads/master:refs/remotes/origin/master}}
 {{ &gt; git config remote.origin.url git@github.com:MyOrganization/my-repo.git # timeout=10}}
 {{ &gt; git config --add remote.origin.fetch +refs/pull/47/head:refs/remotes/origin/PR-47 # timeout=10}}
 {{ &gt; git config --add remote.origin.fetch +refs/heads/master:refs/remotes/origin/master # timeout=10}}
 {{ &gt; git config remote.origin.url git@github.com:MyOrganization/my-repo.git # timeout=10}}
 <tt>Fetching without tags</tt>
 <tt>Fetching upstream changes from git@github.com:MyOrganization/my-repo.git</tt>
 <tt>using GIT_SSH to set credentials MyJenkins@github.com SSH key</tt>
 {{ &gt; git fetch --no-tags --force --progress git@github.com:MyOrganization/my-repo.git +refs/pull/47/head:refs/remotes/origin/PR-47 +refs/heads/master:refs/remotes/origin/master}}
 <tt>Merging remotes/origin/master commit 7e881478aa951a0f9caad966907c68c5c058a935 into PR head commit 5729a1e31f5d0691d16801604312de0af03da009</tt>
 {{ &gt; git config core.sparsecheckout # timeout=10}}
 {{ &gt; git checkout -f 5729a1e31f5d0691d16801604312de0af03da009}}
 {{ &gt; git merge 7e881478aa951a0f9caad966907c68c5c058a935 # timeout=10}}
 {{ &gt; git rev-parse HEAD^ {commit} timeout=10}}
 <tt>Merge succeeded, producing 5729a1e31f5d0691d16801604312de0af03da009</tt>
 <tt>Checking out Revision 5729a1e31f5d0691d16801604312de0af03da009 (PR-47)</tt>
 {{ &gt; git config core.sparsecheckout # timeout=10}}
 {{ &gt; git checkout -f 5729a1e31f5d0691d16801604312de0af03da009}}
 <tt>Commit message: "Merge branch 'master' of <a href="https://github.com/MyOrganization/my-repo" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/MyOrganization/my-repo</a> into feature/search-api-new-branch"</tt>
 <tt>First time build. Skipping changelog.</tt>
 <tt>Running in Durability level: MAX_SURVIVABILITY</tt>
 <tt>Loading library jenkins-pipeline-libraries@master</tt>
 <tt>Examining MyOrganization/jenkins-pipeline-libraries</tt>
 <tt>Attempting to resolve master as a branch</tt>
 <tt>Resolved master as branch master at revision 190ac158caf5abb464702aa7515c417f01143052</tt>
 <tt>using credential github-ci-username-password</tt>
 <tt>Cloning the remote Git repository</tt>
 <tt>Cloning with configured refspecs honoured and without tags</tt>
 <tt>Cloning repository <a href="https://github.com/MyOrganization/jenkins-pipeline-libraries.git" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/MyOrganization/jenkins-pipeline-libraries.git</a></tt>
 {{ &gt; git init /var/jenkins_home/workspace/my-repo_PR-47@libs/jenkins-pipeline-libraries # timeout=10}}
 <tt>Fetching upstream changes from <a href="https://github.com/MyOrganization/jenkins-pipeline-libraries.git" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/MyOrganization/jenkins-pipeline-libraries.git</a></tt>
 {{ &gt; git --version timeout=10}}
 {{using GIT_ASKPASS to set credentials }}
 {{ &gt; git fetch --no-tags --force --progress <a href="https://github.com/MyOrganization/jenkins-pipeline-libraries.git" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/MyOrganization/jenkins-pipeline-libraries.git</a> +refs/heads/master:refs/remotes/origin/master}}
 {{ &gt; git config remote.origin.url <a href="https://github.com/MyOrganization/jenkins-pipeline-libraries.git" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/MyOrganization/jenkins-pipeline-libraries.git</a> # timeout=10}}
 {{ &gt; git config --add remote.origin.fetch +refs/heads/master:refs/remotes/origin/master timeout=10}}
 {{ &gt; git config remote.origin.url <a href="https://github.com/MyOrganization/jenkins-pipeline-libraries.git" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/MyOrganization/jenkins-pipeline-libraries.git</a> # timeout=10}}
 <tt>Fetching without tags</tt>
 <tt>Fetching upstream changes from <a href="https://github.com/MyOrganization/jenkins-pipeline-libraries.git" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/MyOrganization/jenkins-pipeline-libraries.git</a></tt>
 {{using GIT_ASKPASS to set credentials }}
 {{ &gt; git fetch --no-tags --force --progress <a href="https://github.com/MyOrganization/jenkins-pipeline-libraries.git" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/MyOrganization/jenkins-pipeline-libraries.git</a> +refs/heads/master:refs/remotes/origin/master}}
 <tt>Checking out Revision 190ac158caf5abb464702aa7515c417f01143052 (master)</tt>
 {{ &gt; git config core.sparsecheckout # timeout=10}}
 {{ &gt; git checkout -f 190ac158caf5abb464702aa7515c417f01143052}}
 <tt>Commit message: "Fix typo"</tt>
 <tt>First time build. Skipping changelog.</tt>
 <tt><span class="error">&#91;Pipeline&#93;</span> Start of Pipeline</tt>
 <tt>........ Lots of stuff .....</tt></p>

<p>I figure it somehow still believes that <tt>7e88</tt> is master HEAD (based on <tt>Merging remotes/origin/master commit 7e881478aa951a0f9caad966907c68c5c058a935 into PR head commit 5729a1e31f5d0691d16801604312de0af03da009</tt>). It printed one correct reference to the true master HEAD (<tt>76ff</tt>) in the 3rd build log line.</p>
</pre>
</details>

</ul>
</details>

<p>Thank you!</p>

<p>Scan log:</p>

<p><tt>...</tt><br/>
<tt>Checking pull request #47</tt></p>

<p><tt>‘Jenkinsfile’ found Invalid github merge_commit_sha for pull request 47 : Head and base commits do match merge commit 5729a1e31f5d0691d16801604312de0af03da009+7e881478aa951a0f9caad966907c68c5c058a935 (567c876ca2e53637b53edd7c4eedca77af15aa65), expected head and base 76ff9c62d27d6b60666c08d0686e3100974fd1fe+5729a1e31f5d0691d16801604312de0af03da009 Met criteria Changes detected: PR-47 (5729a1e31f5d0691d16801604312de0af03da009+7e881478aa951a0f9caad966907c68c5c058a935 (567c876ca2e53637b53edd7c4eedca77af15aa65) → 5729a1e31f5d0691d16801604312de0af03da009+7e881478aa951a0f9caad966907c68c5c058a935 (UNKNOWN_MERGE_STATE)) Scheduled build for branch: PR-47</tt></p>

<p>Build log (it now starts building the pipeline):</p>

<p><tt>Branch indexing</tt><br/>
 <tt>21:04:15 Connecting to <a href="https://api.github.com/" class="external-link" target="_blank" rel="nofollow noopener">https://api.github.com</a> using ci@​mydomain.com/******</tt><br/>
 <tt>Invalid github merge_commit_sha for pull request 47 : Head and base commits do match merge commit 5729a1e31f5d0691d16801604312de0af03da009+7e881478aa951a0f9caad966907c68c5c058a935 (567c876ca2e53637b53edd7c4eedca77af15aa65), expected head and base 76ff9c62d27d6b60666c08d0686e3100974fd1fe+5729a1e31f5d0691d16801604312de0af03da009</tt><br/>
 <tt>Checking out git git@​github.com:MyOrganization/my-repo.git into /var/jenkins_home/workspace/my-repo_PR-47@​script to read Jenkinsfile</tt><br/>
 <tt>using credential github-sshkey</tt><br/>
 <tt>Cloning the remote Git repository</tt><br/>
 <tt>Cloning with configured refspecs honoured and without tags</tt><br/>
 <tt>Cloning repository git@​github.com:MyOrganization/my-repo.git</tt><br/>
 {{ > git init /var/jenkins_home/workspace/my-repo_PR-47@​script # timeout=10}}<br/>
 <tt>Fetching upstream changes from git@​github.com:MyOrganization/my-repo.git</tt><br/>
 {{ > git --version # timeout=10}}<br/>
 <tt>using GIT_SSH to set credentials MyJenkins@​github.com SSH key</tt><br/>
 {{ > git fetch --no-tags --force --progress git@​github.com:MyOrganization/my-repo.git +refs/pull/47/head:refs/remotes/origin/PR-47 +refs/heads/master:refs/remotes/origin/master}}<br/>
 {{ > git config remote.origin.url git@​github.com:MyOrganization/my-repo.git # timeout=10}}<br/>
 {{ > git config --add remote.origin.fetch +refs/pull/47/head:refs/remotes/origin/PR-47 # timeout=10}}<br/>
 {{ > git config --add remote.origin.fetch +refs/heads/master:refs/remotes/origin/master # timeout=10}}<br/>
 {{ > git config remote.origin.url git@​github.com:MyOrganization/my-repo.git # timeout=10}}<br/>
 <tt>Fetching without tags</tt><br/>
 <tt>Fetching upstream changes from git@​github.com:MyOrganization/my-repo.git</tt><br/>
 <tt>using GIT_SSH to set credentials MyJenkins@​github.com SSH key</tt><br/>
 {{ > git fetch --no-tags --force --progress git@​github.com:MyOrganization/my-repo.git +refs/pull/47/head:refs/remotes/origin/PR-47 +refs/heads/master:refs/remotes/origin/master}}<br/>
 <tt>Merging remotes/origin/master commit 7e881478aa951a0f9caad966907c68c5c058a935 into PR head commit 5729a1e31f5d0691d16801604312de0af03da009</tt><br/>
 {{ > git config core.sparsecheckout # timeout=10}}<br/>
 {{ > git checkout -f 5729a1e31f5d0691d16801604312de0af03da009}}<br/>
 {{ > git merge 7e881478aa951a0f9caad966907c68c5c058a935 # timeout=10}}<br/>
 {{ > git rev-parse HEAD^ {commit} timeout=10}}<br/>
 <tt>Merge succeeded, producing 5729a1e31f5d0691d16801604312de0af03da009</tt><br/>
 <tt>Checking out Revision 5729a1e31f5d0691d16801604312de0af03da009 (PR-47)</tt><br/>
 {{ > git config core.sparsecheckout # timeout=10}}<br/>
 {{ > git checkout -f 5729a1e31f5d0691d16801604312de0af03da009}}<br/>
 <tt>Commit message: "Merge branch 'master' of <a href="https://github.com/MyOrganization/my-repo" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/MyOrganization/my-repo</a> into feature/search-api-new-branch"</tt><br/>
 <tt>First time build. Skipping changelog.</tt><br/>
 <tt>Running in Durability level: MAX_SURVIVABILITY</tt><br/>
 <tt>Loading library jenkins-pipeline-libraries@​master</tt><br/>
 <tt>Examining MyOrganization/jenkins-pipeline-libraries</tt><br/>
 <tt>Attempting to resolve master as a branch</tt><br/>
 <tt>Resolved master as branch master at revision 190ac158caf5abb464702aa7515c417f01143052</tt><br/>
 <tt>using credential github-ci-username-password</tt><br/>
 <tt>Cloning the remote Git repository</tt><br/>
 <tt>Cloning with configured refspecs honoured and without tags</tt><br/>
 <tt>Cloning repository <a href="https://github.com/MyOrganization/jenkins-pipeline-libraries.git" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/MyOrganization/jenkins-pipeline-libraries.git</a></tt><br/>
 {{ > git init /var/jenkins_home/workspace/my-repo_PR-47@​libs/jenkins-pipeline-libraries # timeout=10}}<br/>
 <tt>Fetching upstream changes from <a href="https://github.com/MyOrganization/jenkins-pipeline-libraries.git" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/MyOrganization/jenkins-pipeline-libraries.git</a></tt><br/>
 {{ > git --version timeout=10}}<br/>
 {{using GIT_ASKPASS to set credentials }}<br/>
 {{ > git fetch --no-tags --force --progress <a href="https://github.com/MyOrganization/jenkins-pipeline-libraries.git" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/MyOrganization/jenkins-pipeline-libraries.git</a> +refs/heads/master:refs/remotes/origin/master}}<br/>
 {{ > git config remote.origin.url <a href="https://github.com/MyOrganization/jenkins-pipeline-libraries.git" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/MyOrganization/jenkins-pipeline-libraries.git</a> # timeout=10}}<br/>
 {{ > git config --add remote.origin.fetch +refs/heads/master:refs/remotes/origin/master timeout=10}}<br/>
 {{ > git config remote.origin.url <a href="https://github.com/MyOrganization/jenkins-pipeline-libraries.git" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/MyOrganization/jenkins-pipeline-libraries.git</a> # timeout=10}}<br/>
 <tt>Fetching without tags</tt><br/>
 <tt>Fetching upstream changes from <a href="https://github.com/MyOrganization/jenkins-pipeline-libraries.git" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/MyOrganization/jenkins-pipeline-libraries.git</a></tt><br/>
 {{using GIT_ASKPASS to set credentials }}<br/>
 {{ > git fetch --no-tags --force --progress <a href="https://github.com/MyOrganization/jenkins-pipeline-libraries.git" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/MyOrganization/jenkins-pipeline-libraries.git</a> +refs/heads/master:refs/remotes/origin/master}}<br/>
 <tt>Checking out Revision 190ac158caf5abb464702aa7515c417f01143052 (master)</tt><br/>
 {{ > git config core.sparsecheckout # timeout=10}}<br/>
 {{ > git checkout -f 190ac158caf5abb464702aa7515c417f01143052}}<br/>
 <tt>Commit message: "Fix typo"</tt><br/>
 <tt>First time build. Skipping changelog.</tt><br/>
 <tt><span class="error">&#91;Pipeline&#93;</span> Start of Pipeline</tt><br/>
 <tt>........ Lots of stuff .....</tt></p>

<p>I figure it somehow still believes that <tt>7e88</tt> is master HEAD (based on <tt>Merging remotes/origin/master commit 7e881478aa951a0f9caad966907c68c5c058a935 into PR head commit 5729a1e31f5d0691d16801604312de0af03da009</tt>). It printed one correct reference to the true master HEAD (<tt>76ff</tt>) in the 3rd build log line.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>audunhalland:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-57371?focusedId=367278&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-367278">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Sorry, I had some formatting problems pasting the logs into the previous comment.</p>
</pre>
</details>

</ul>
</details>

<p>Sorry, I had some formatting problems pasting the logs into the previous comment.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/bitwiseman.png" title="bitwiseman's avatar" /> bitwiseman:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-57371?focusedId=367361&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-367361">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=audunhalland" class="user-hover" rel="audunhalland">audunhalland</a>
Formatting: no problem.  
Yes, one part of Jenkins thinks that  definitely believes that the head of master is <tt>7e88</tt>, when it is actually <tt>76ff</tt>.  But it is unclear <em>how</em> that is happening. </p>

<p>It sounds like the patch has unblocked you.  I'm working on finalizing a new version that includes this change.</p>
</pre>
</details>

</ul>
</details>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=audunhalland" class="user-hover" rel="audunhalland">audunhalland</a><br/>
Formatting: no problem.  <br/>
Yes, one part of Jenkins thinks that  definitely believes that the head of master is <tt>7e88</tt>, when it is actually <tt>76ff</tt>.  But it is unclear <em>how</em> that is happening. </p>

<p>It sounds like the patch has unblocked you.  I'm working on finalizing a new version that includes this change.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/bitwiseman.png" title="bitwiseman's avatar" /> bitwiseman:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-57371?focusedId=367573&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-367573">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=audunhalland" class="user-hover" rel="audunhalland">audunhalland</a> 
v2.5.3 should be out in the next day or so.  If you want to help improve the quality please try this new patch: 
<a href="https://repo.jenkins-ci.org/incrementals/org/jenkins-ci/plugins/github-branch-source/2.5.3-rc840.185f1729c3a6/" class="external-link" target="_blank" rel="nofollow noopener">https://repo.jenkins-ci.org/incrementals/org/jenkins-ci/plugins/github-branch-source/2.5.3-rc840.185f1729c3a6/</a></p>


</pre>
</details>

</ul>
</details>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=audunhalland" class="user-hover" rel="audunhalland">audunhalland</a> <br/>
v2.5.3 should be out in the next day or so.  If you want to help improve the quality please try this new patch: <br/>
<a href="https://repo.jenkins-ci.org/incrementals/org/jenkins-ci/plugins/github-branch-source/2.5.3-rc840.185f1729c3a6/" class="external-link" target="_blank" rel="nofollow noopener">https://repo.jenkins-ci.org/incrementals/org/jenkins-ci/plugins/github-branch-source/2.5.3-rc840.185f1729c3a6/</a></p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>audunhalland:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-57371?focusedId=367764&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-367764">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Thanks, excited to try it out. Though our famous PR-47 is already long gone, I'll be sure to keep my eyes open for failing PRs in the future!</p>

<p>Thank you again for improving Jenkins and for great support on this issue!</p>
</pre>
</details>

</ul>
</details>

<p>Thanks, excited to try it out. Though our famous PR-47 is already long gone, I'll be sure to keep my eyes open for failing PRs in the future!</p>

<p>Thank you again for improving Jenkins and for great support on this issue!</p>
